presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/MergeAdjacentWindows.java

```diff
@@ -46,7 +47,16 @@
     @Override
     public Optional<PlanNode> apply(WindowNode parent, Captures captures, Context context)
     {
-        WindowNode child = captures.get(CHILD);
+        // Pulling the descendant WindowNode above projects is done as a part of this rule, as opposed in a
+        // separate rule, because that pullup is not useful on its own, and could be undone by other rules.
+        // For example, a rule could insert a project-off node between adjacent WindowNodes that use different
+        // input symbols.
```

Comment:
Ok. Thank you!

Let make it this way. First of all do not use `Lookup::resolve`. as it got deprecated. I suggest to make this rule to return a pattern like:
```
 - window
   - project
     - window
```
I believe this pattern will cover most of use cases, hoping that iterative optimizer could merge several simple projections into one.

Rules should not traverse the plan recursively. Currently it only works because `Memo` stores only a single node per group, which is going to change.

What about other nodes which could be skipped during `WindowNode` merging, like `DistintNode`, `FilterNode`?